### PR TITLE
refactored UpdateReleaseNotes

### PR DESCRIPTION
**Pull request by epeee**
@wwilk a very first version of the refactoring we were talking about in one of the last pr's.

Javadoc is still missing, also the naming of certain classes might change and also visibility. I just wanted to get your feedback.
To sum it up, it is more code now but it is also more separated and reusable.

### CODE REVIEW

File: subprojects/shipkit/src/main/groovy/org/shipkit/internal/gradle/notes/tasks/CollectionToMapTransformer.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+public class CollectionToMapTransformer<A extends Collection<B>, B, K, V> {
+
+    private final Function<B, Map.Entry<K, V>> transformFunction;
+
+    public CollectionToMapTransformer(Function<B, Map.Entry<K, V>> transformFunction) {
+        this.transformFunction = transformFunction;
+    }
+
+    public Map<K, V> transform(A collection) {
+        Map<K, V> contributorMap = new HashMap<K, V>();
```
[the remark this answers is not in the document]
Comment (mockitoguy) [reply subprojects/shipkit/src/main/groovy/org/shipkit/internal/gradle/notes/tasks/CollectionToMapTransformer.java]: question: would it all be simpler if we moved to Java8?

[the remark this answers is not in the document]
Comment (wwilk) [reply subprojects/shipkit/src/main/groovy/org/shipkit/internal/gradle/notes/tasks/CollectionToMapTransformer.java]: Sure, lambdas would make it simpler.

**Comment by wwilk** [comment]
I think complexity might be increased in general. On the other hand at the high level in UpdateReleaseNotes it was decreased due to usage of builder. So I like the fact that this complexity was removed from UpdateReleaseNotes, but that's true that it's a lot of code for a simple thing, and not really a lot of this code we can reuse. So I'm neutral in this case, whatever you guys decide.

**Comment by epeee** [comment]
Yes, complexity increased a little bit due to the builder. On the other hand this abstraction allows us to decouple the map generation in a clean way from the UpdateReleaseNotes. The builder also generates ~half of the new lines of code.
I'm ok with not merging it, I just thought it would make UpdateReleaseNotes a little bit cleaner/separate certain logic and therefore make it easier to test.

File: subprojects/shipkit/src/main/groovy/org/shipkit/internal/gradle/notes/tasks/CollectionToMapTransformer.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (epeee) [reply subprojects/shipkit/src/main/groovy/org/shipkit/internal/gradle/notes/tasks/CollectionToMapTransformer.java]: Good catch! thx!

**Comment by mockitoguy** [comment]
@mockito/shipkit-developers, we need one more opinion, who is going to be a tie-breaker?

If there is no other opinion, I will consider my vote as +1.01 and close the PR without merging. Good discussion!

**Comment by epeee** [comment]
yep, let's close this one and go for less code.

**Pull request closed without merging**